lib/app_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,45 @@
+defmodule AppWeb.Auth do
+  use AppWeb, :controller
+  # use AppWeb, :router
+  import Plug.Conn
+  # alias App.Repo
+
+  def init(opts), do: opts
+
+  def call(conn, _opts) do
+    person_id = get_session(conn, :person_id)
+    cond do
+      person = conn.assigns[:current_person] ->
+        assign(conn, :current_person, person)
+
+      person = person_id && App.Ctx.get_person!(person_id) ->
+        assign(conn, :current_person, person)
+
+      true ->
+        assign(conn, :current_person, nil)
+    end
+  end
+
+  def login(conn, person) do
+    conn
+    |> assign(:current_person, person)
+    |> put_session(:person_id, person.id)
+    |> configure_session(renew: true)
+  end
+
+  def logout(conn) do
+    configure_session(conn, drop: true)
```

Comment:
My reading of this function is that it will only drop the session for _one_ browser tab. 
If the user has multiple and only performs logout in one, will their session still be valid in the others?

Reply:
My understanding is that the function will drop the session for the browser (ie all the tabs). I'll double check this on my localhost.